api/handlers.go

```diff
@@ -890,8 +890,28 @@ func getEvents(c *context, w http.ResponseWriter, r *http.Request) {
 		f.Flush()
 	}
 
-	c.eventsHandler.Add(r.RemoteAddr, w)
-	c.eventsHandler.Wait(r.RemoteAddr, until)
+	eventsChan, cancelFunc := c.watchQueue.Watch()
+	defer cancelFunc()
+	// TODO(nishanttotla): more code needed for --until
+	fmt.Println("UNTIL", until)
+	for {
+		select {
+		case eChan := <-eventsChan:
+			if e, ok := eChan.(*cluster.Event); ok {
+				fmt.Println("NEW HANDLE CALLING NORMALIZE EVENT", e.Time, e.Action, e.From)
+				if data, err := normalizeEvent(e); err != nil {
+				} else {
+					_, err := fmt.Fprint(w, string(data))
+					if f, ok := w.(http.Flusher); ok {
+						f.Flush()
+					}
+					if err != nil {
+						log.Debugf("failed to write event to output stream %s", err.Error())
```

Comment:
If there was an error writing to the socket, this code should return from the function. Currently this loop will keep receiving events and trying to write them forever.